api/src/test/java/org/pmiops/workbench/actionaudit/auditors/SumoLogicAuditorTest.java

```diff
@@ -0,0 +1,216 @@
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+public class SumoLogicAuditorTest {
+
+  private static final long USER_ID = 1L;
+  private static final String USER_EMAIL = "user@researchallofus.org";
+  private static final long WORKSPACE_ID = 1L;
+  private static final String WORKSPACE_NAMESPACE = "aou-rw-test-c7dec260";
+  private static final String WORKSPACE_FIRECLOUD_NAME = "mytestworkspacename";
+
+  private static final String EGRESS_EVENT_PROJECT_NAME = WORKSPACE_NAMESPACE;
+  private static final String EGRESS_EVENT_VM_NAME = "all-of-us-" + USER_ID + "-m";
+
+  // Pre-built data objects for test.
+  private DbUser dbUser;
+  private DbWorkspace dbWorkspace;
+  private List<UserRole> firecloudUserRoles = new ArrayList<>();
+
+  @Autowired private SumoLogicAuditor sumoLogicAuditor;
+
+  @MockBean private ActionAuditService mockActionAuditService;
+  @MockBean private WorkspaceService mockWorkspaceService;
+  @MockBean private UserDao mockUserDao;
+
+  @Captor private ArgumentCaptor<Collection<ActionAuditEvent>> eventsCaptor;
+
+  @Rule public final ExpectedException exception = ExpectedException.none();
+
+  @Before
+  public void setUp() {
+    dbUser = new DbUser();
+    dbUser.setUserId(USER_ID);
+    dbUser.setUsername(USER_EMAIL);
+    when(mockUserDao.findUserByUsername(USER_EMAIL)).thenReturn(dbUser);
+
+    dbWorkspace = new DbWorkspace();
+    dbWorkspace.setWorkspaceId(WORKSPACE_ID);
+    dbWorkspace.setWorkspaceNamespace(WORKSPACE_NAMESPACE);
+    dbWorkspace.setFirecloudName(WORKSPACE_FIRECLOUD_NAME);
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(dbWorkspace);
+    firecloudUserRoles.add(new UserRole().email(USER_EMAIL));
+    when(mockWorkspaceService.getFirecloudUserRoles(WORKSPACE_NAMESPACE, WORKSPACE_FIRECLOUD_NAME))
+        .thenReturn(firecloudUserRoles);
+  }
+
+  Set<String> extractValuesFromEvents(
+      Collection<ActionAuditEvent> events, Function<ActionAuditEvent, String> fn) {
+    return events.stream().map(fn).collect(Collectors.toSet());
+  }
+
+  @Test
+  public void testFireEgressEvent() {
+    sumoLogicAuditor.fireEgressEvent(
+        new EgressEvent()
+            .projectName(EGRESS_EVENT_PROJECT_NAME)
+            .vmName(EGRESS_EVENT_VM_NAME)
+            .timeWindowStart(0l)
+            .egressMib(12.3));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Ensure all events have the expected set of constant fields.
+    assertThat(events.stream().map(event -> event.getAgentType()).collect(Collectors.toSet()))
+        .containsExactly(AgentType.USER);
+    assertThat(events.stream().map(event -> event.getActionType()).collect(Collectors.toSet()))
+        .containsExactly(ActionType.HIGH_EGRESS);
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
```

Comment:
You're doing enough of this that it could be helpful to have an assertion helper funciton in the parent class like `assertContainsUniqueValue(Collection<T> objects, Function<T, U> extractor, U value);` I Could use that in a few places as well.

Reply:
Yeah... I almost started doing that, but it only would have saved a "bit" of code, and would add indirection which has a cost on readability. I'll leave as-is, but I agree some helpers may be warranted if this grows in size.